pkg/addons/registry/registry.go

```diff
@@ -120,6 +121,19 @@ func (o *Registry) GenerateHelmConfig(onlyDefaults bool) ([]v1beta1.Chart, []v1b
 		URL:  ChartURL,
 	}
 
+	// use a static cluster IP for the registry service based on the cluster CIDR range
+	serviceCIDR := v1beta1.DefaultNetwork().ServiceCIDR
+	if o.config.Spec != nil && o.config.Spec.Network != nil {
+		serviceCIDR = o.config.Spec.Network.ServiceCIDR
+	}
+	registryServiceIP, err := helpers.GetLowerBandIP(serviceCIDR, 10)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get cluster IP for registry service: %w", err)
+	}
```

Comment:
Right now this would pull the IP at index 10, so for the k0s default service CIDR of `10.96.0.0/12` would be `10.96.0.11` and follow a similar patter for different ranges. Is this sufficient to specify a static "index" like this? Do we need to make it dynamic at this time? If so, how could we go about that?

Reply:
this seems good to me
if we have a collision in the future, we can deal with it then